crates/ratchet-core/src/ops/cast.rs

```diff
@@ -17,25 +17,41 @@ pub struct Cast {
     dst_dt: DType,
 }
 
-impl Cast {
-    fn register_bindings<SP: WgslPrimitive, DP: WgslPrimitive>(
+impl KernelRenderable for CastKernels {
+    fn register_bindings<SP: WgslPrimitive>(
         &self,
         builder: &mut WgslKernelBuilder,
         _: bool,
     ) -> Result<(), OperationError> {
         builder.register_storage("X", BindingMode::ReadOnly, Array::<SP>::default());
-        builder.register_storage("Y", BindingMode::ReadWrite, Array::<DP>::default());
+        let inner = match self {
+            CastKernels::Standard(s) => s,
+        };
+        //TODO: This is a bit of a hack, this match should be formalized
+        let dst_accessor = match SP::W {
```

Comment:
@ivarflakstad any ideas on some type wizardry I can do here?
I know the width of `SP` (Source Primitive), and I know the `dst_dt` as a value.
But tricky to actually make `Array::<DP>`